src/inc/ComHelper.hpp

```diff
@@ -57,7 +57,7 @@ namespace MSIX {
     {
         bool Matches(REFIID riid, void** ppvObject)
         {
-            if(ChainInterfaces<I0, Chain...>::Includes(riid))
+            if(riid == UuidOfImpl<I0>::iid || ChainInterfaces<I0, Chain...>::Includes(riid))

```

Comment:
>riid == UuidOfImpl<I0>::iid [](start = 15, length = 27)

I would do this in the 2 type specialization, so it will hold true for every call.